app/main.go

```diff
@@ -96,9 +105,62 @@ func main() {
 		log.Fatalf("Invalid flags: %v", err)
 	}
 
-	http.HandleFunc("/", index)
-	http.HandleFunc("/sync", sync)
-	http.HandleFunc("/cleanup", cleanup)
+	config := tsbridge.NewConfig(&tsbridge.ConfigOptions{
+		Filename:                 *metricConfig,
+		MinPointAge:              *minPointAge,
+		CounterResetInterval:     *counterResetInterval,
+		SDLookBackInterval:       *sdLookBackInterval,
+		SDInternalMetricsProject: *sdInternalMetricsProject,
+		UpdateParallelism:        *updateParallelism,
+		EnableStatusPage:         *enableStatusPage,
+		StorageEngine:            *storageEngine,
+		SyncPeriod:               *syncPeriod,
+		SyncCleanupAfter:         *syncCleanupAfter,
+	})
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		index(w, r, config)
+	})
+	http.HandleFunc("/sync", func(w http.ResponseWriter, r *http.Request) {
+		syncHandler(w, r, config)
+	})
+	http.HandleFunc("/cleanup", func(w http.ResponseWriter, r *http.Request) {
+		cleanupHandler(w, r, config)
+	})
+
+	// Run a sync loop for standalone use
+	if env.NotAppEngine() {
+		log.Debug("Running outside of appengine, starting up a sync loop...")
+		ctx, cancel := context.WithCancel(context.Background())
+		count := 0
+		go func() {
+			defer cancel()
+			for {
+				select {
+				case <-time.After(config.Options.SyncPeriod):
+					ctx, cancel := context.WithTimeout(ctx, *updateTimeout)
+					log.WithContext(ctx).Debugf("Running sync...")
+					if err := sync(ctx, config); err != nil {
+						log.WithContext(ctx).Errorf("error running sync() routine: %v", err)
+						return
+					}
+					// perform a cleanup every 100th cycle
+					if count%config.Options.SyncCleanupAfter == 0 {
```

Comment:
Perhaps using a separate timer for cleanup will be simpler.

Reply:
Not really since BoltDB supports only 1 writer at the time. Then we need to use locks to prevent race conditions and do a lot of brittle timing, no?